libbpf-rs/src/object.rs

```diff
@@ -327,7 +327,9 @@ impl Object {
 
             // Get the map fd
             let fd = unsafe { libbpf_sys::bpf_map__fd(next_ptr) };
-            if fd < 0 {
+            // Old kernels can have uninitialized internal maps:
+            // https://github.com/libbpf/libbpf/blob/5c31bcf220f66e70f39fd141f5b0c55c6ab65e8e/src/libbpf.c#L5009-L5022
+            if fd < 0 && !unsafe { libbpf_sys::bpf_map__is_internal(next_ptr) } {
```

Comment:
Do we also need the "!kernel_supports(obj, FEAT_GLOBAL_DATA))" check here as well?

Reply:
That would be ideal. I'm afraid these bits are not exposed by libbpf (`kernel_supports` itself and the array it queries are `static`).